clusters.cnp.example.org/definition.yaml

```diff
@@ -6,17 +6,14 @@ metadata:
 spec:
   defaultCompositionRef:
     name: clusters.cnp.example.org
-  # This policy is equivalent to the existing InfrastructurePublication. Making
-  # this a field of the InfrastructureDefinition implies that the package author
-  # controls the publication of packaged types; it cannot easily be overridden
-  # without forking/duplicating the package. The options are 'Private' (the
-  # default), or 'Published'.
-  publicationPolicy: Published
-  # Publishing a composite resource defines a namespaced proxy of the composite
-  # resource. The group, version, and schema of this resource is identical to
-  # those of the composite resource but its kind must differ. Conceptually this
-  # is a 'published composite resource', rather than 'a requirement'.
-  publishedNames:
+  # Composite resources are cluster scoped, but each composite resource may
+  # optionally support a namespaced 'composite resource requirement'. A
+  # requirement acts as a proxy for its composite resource - changes made to the
+  # requirement are reflected on the composite resource. A requirement shares
+  # the API group and schema of its corresponding composite resouce, but must
+  # have a different name. Omit the requirement names when defining a composite
+  # resource that should not support a corresponding requirement resource.
+  requirementNames:
```

Comment:
Does this generate Kind: ClusterRequirement. I also think Claim is a very interesting idea instead of Requirement as the user would interact with a ClusterClaim.

Also think this is an interesting feature, but making users name them to enable them instead of it being an override of a default seems like a miss maybe from DX.

I'll defer though final decision to @negz